engine/src/main/java/org/hibernate/search/query/dsl/impl/ConnectedMultiFieldsTermQueryBuilder.java

```diff
@@ -107,19 +113,17 @@ private Query createQuery(FieldContext fieldContext, ConversionContext conversio
 		final String searchTerm = buildSearchTerm( fieldContext, documentBuilder, conversionContext );
 
 		if ( !applyTokenization || BridgeAdaptorUtils.unwrapAdaptorOnly( fieldBridge, IgnoreAnalyzerBridge.class ) != null ) {
-			perFieldQuery = createTermQuery( fieldContext, searchTerm );
+			return createTermQuery( fieldContext, searchTerm );
+		}
+
+		// we need to build differentiated queries depending of if the search terms should be analyzed
```

Comment:
Since it's you, I'm sure you'll understand :D : could you please replace "of if" with "on if" while you're at it?